src/Parser.php

```diff
@@ -95,10 +95,12 @@ public function parse($input, $file = null)
             }
 
             if ($feature && $node instanceof FeatureNode) {
+                // @codeCoverageIgnoreStart
```

Comment:
I could not replicate this and the other cases below. In particular, some seem to be impossible e.g. given that we only processing up to 4 different types of tokens, we should not get any other "unexpected" token unless the lexer is broken and lets such a token through.

Reply:
I tried testing it again, this time by mocking the lexer to return my own set of tokens:
1. `['type' => 'Feature'], ['type' => 'Feature'], ['type' => 'EOS']` - "Scenario, ... expected but Feature found"
2. ['type' => 'Feature'], ['type' => 'EOS'], ['type' => 'Feature'], ['type' => 'EOS'] - no error, because it stops on first `EOS`.

In short, even if the lexer misbehaves, we won't handle more than one feature because:
1. `parse()` calls `parseExpression()` until `EOS` is encountered
2. `parseExpression()` calls `parseFeature()`
3. `parseFeature()` goes into a loop until `EOS` is encountered
4. therefore `parse()` is never called again (and when it receives back control, we're in EOS so it terminates).